### PR TITLE
[#1758] feat(autoIncrement) : ColumnChange support auto_increment.

### PR DESCRIPTION
**Pull request by Clearvive**
### What changes were proposed in this pull request?

ColumnChange support auto_increment.
Including `AddColumn` and `UpdateColumnAutoIncrement`

### Why are the changes needed?
Fix: #1758 

### Does this PR introduce _any_ user-facing change?
Add tableChange

### How was this patch tested?
UT+IT

### CODE REVIEW

File: catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java [hunk 1]
```diff
@@ -478,6 +478,10 @@ private String addColumnFieldDefinition(TableChange.AddColumn addColumn) {
         .append(dataType)
         .append(SPACE);
 
+    if (addColumn.isAutoIncrement()) {
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java:481]: shoud we types check here?

File: catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/ops/IcebergTableOpsHelper.java [hunk 2]
```diff
@@ -191,6 +191,10 @@ private void doAddColumn(
       parentStruct = icebergTableSchema.asStruct();
     }
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/ops/IcebergTableOpsHelper.java:192]: miss  TableChange.UpdateColumnAutoIncrement check

File: catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveCatalogOperations.java [hunk 3]
```diff
@@ -931,6 +931,9 @@ private void doRemoveProperty(
 
   private void doAddColumn(List<FieldSchema> cols, TableChange.AddColumn change) {
```
Comment (FANNG1) [inline catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveCatalogOperations.java:932]: miss  TableChange.UpdateColumnAutoIncrement check

**Comment by FANNG1** [comment]
@mchades please help to review

File: api/src/main/java/com/datastrato/gravitino/rel/TableChange.java [hunk 4]
```diff
@@ -1285,4 +1335,56 @@ public int hashCode() {
       return result;
     }
   }
+
+  /** A TableChange to update the autoIncrement of a field. */
```
Comment (FANNG1) [inline api/src/main/java/com/datastrato/gravitino/rel/TableChange.java:1339]: please add the description about `autoIncrement`,  true is to add autoIncrement, false is to delete autoIncrement.

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/catalog/jdbc/mysql/CatalogMysqlIT.java [hunk 5]
```diff
@@ -1382,4 +1382,90 @@ void testOperationTableIndex() {
...
+
+    Table table = tableCatalog.loadTable(tableIdentifier);
+
+    Column col6 = Column.of("col_6", Types.LongType.get(), "id", false, true, null);
+    Index[] indices = new Index[] {Indexes.createMysqlPrimaryKey(new String[][] {{"col_6"}})};
+    newColumns = new Column[] {col1, col2, col3, col4, col5, col6};
+    assertionsTableInfo(
+        tableName, table_comment, Arrays.asList(newColumns), properties, indices, table);
+
+    // Test update auto_increment col.
```
Comment (FANNG1) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/catalog/jdbc/mysql/CatalogMysqlIT.java:1452]: please add some description about what's the purpose of the test? to make it more readable.

File: catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java [hunk 6]
```diff
@@ -478,6 +512,13 @@ private String addColumnFieldDefinition(TableChange.AddColumn addColumn) {
...
+      Preconditions.checkArgument(
+          Types.allowAutoIncrement(addColumn.getDataType()),
```
Comment (mchades) [inline catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java:517]: Can this simplified to `Types.allowAutoIncrement(addColumn.getDataType() || !addColumn.isAutoIncrement()`?

**Review by mchades — COMMENTED** [review]
Exception tests for Hive and Iceberg is missed

**Review by mchades — APPROVED** [review]
LGTM

**Pull request merged**